swift/Tests/Blockchains/CardanoTests.swift

```diff
@@ -132,4 +132,93 @@ class CardanoTests: XCTestCase {
         let txid = output.txID
         XCTAssertEqual(txid.hexString, "201c537693b005b64a0f0528e366ec67a84be0119ed4363b547f141f2a7770c2")
     }
+
+    func testGetStakingAddress() throws {
+        let stakingAddress = Cardano.getStakingAddress(baseAddress: "addr1q8043m5heeaydnvtmmkyuhe6qv5havvhsf0d26q3jygsspxlyfpyk6yqkw0yhtyvtr0flekj84u64az82cufmqn65zdsylzk23")
+        XCTAssertEqual(stakingAddress, "stake1u80jysjtdzqt88jt4jx93h5lumfr67d273r4vwyasfa2pxcwxllmx")
+    }
+
+    func testSignStakingRegisterAndDelegate() throws {
+        let ownAddress = "addr1q8043m5heeaydnvtmmkyuhe6qv5havvhsf0d26q3jygsspxlyfpyk6yqkw0yhtyvtr0flekj84u64az82cufmqn65zdsylzk23"
+        let stakingAddress = Cardano.getStakingAddress(baseAddress: ownAddress)
+        let poolIdNufi = "7d7ac07a2f2a25b7a4db868a40720621c4939cf6aefbb9a11464f1a6"
+
+        var input = CardanoSigningInput.with {
+            $0.transferMessage.toAddress = ownAddress
+            $0.transferMessage.changeAddress = ownAddress
+            $0.transferMessage.amount = 4000000
```

Comment:
This `amount` looks confused to me, there is another 2ADA `depositAmount`, can we hide this detail for the client?

Reply:
The tx needs an amount, but as maxAmount option is used, the amount does not really matter.  I've added comment to the test code to mark this.
The 2 ADA deposit is set in stone, but I did not want to hardcode in wallet-core code, should come from above.